File: app/src/main/java/com/digzdigital/divinitytoday/data/wp/WpHelper.java
package com.digzdigital.divinitytoday.data.wp;

import android.content.Context;

import com.digzdigital.divinitytoday.data.model.Devotional;

import java.util.ArrayList;

public interface WpHelper {
    void setWpListener(WpListener listener);

    void queryForOnlinePosts(int offset);

    void provideContext(Context context);


    interface WpListener {
        void onPostsLoaded(ArrayList<Devotional> devotionals);

        void onError(String error);
    }
}

File: app/src/main/java/com/digzdigital/divinitytoday/data/wp/WpHelper.kt
package com.digzdigital.divinitytoday.data.wp

import com.digzdigital.divinitytoday.data.model.Devotional
import io.reactivex.Observable

interface WpHelper {

    fun getDevotionals(offset: String, per_page: String = "7"): Observable<List<Devotional>>
}
